soroban-sdk/src/deploy.rs

```diff
@@ -54,85 +57,76 @@ impl Deployer {
     }
 
     /// Get a deployer that deploys contracts that derive their contract IDs
-    /// from the current contract and the provided salt.
-    pub fn with_current_contract(
-        &self,
-        salt: &impl IntoVal<Env, BytesN<32>>,
-    ) -> DeployerWithCurrentContract {
-        let env = self.env();
-        DeployerWithCurrentContract {
-            env: env.clone(),
-            salt: salt.into_val(env),
+    /// from the current contract.
+    pub fn with_current_contract(&self) -> DeployerWithAddress {
+        DeployerWithAddress {
+            env: self.env.clone(),
+            address: self.env.current_contract_address(),
         }
     }
 
-    #[doc(hidden)]
-    /// Get a deployer for contracts that derive their contract IDs from the
-    /// given contract ID and the provided salt.
-    pub fn with_other_contract(
-        &self,
-        contract_id: &Address,
-        salt: &impl IntoVal<Env, BytesN<32>>,
-    ) -> DeployerWithOtherContract {
-        let env = self.env();
-        DeployerWithOtherContract {
-            env: env.clone(),
-            contract_id: contract_id.clone(),
-            salt: salt.into_val(env),
+    /// Get a deployer that deploys contracts that derive their contract IDs
+    /// from the provided address.
+    ///
+    /// The deployer address must authorize all the deployments.
+    pub fn with_address(&self, address: Address) -> DeployerWithAddress {
+        DeployerWithAddress {
+            env: self.env.clone(),
+            address,
         }
     }
+
+    /// Deploys a new instance of Stellar Asset Contract corresponding to
+    /// provided serialized asset.
+    ///
+    /// `serialized_asset` is the Stellar `Asset` XDR serialized to bytes. Refer
+    /// to `stellar-xdr` create for the exact definition.
+    pub fn deploy_stellar_asset(&self, serialized_asset: impl IntoVal<Env, Bytes>) -> Address {
+        self.env
+            .create_asset_contract(serialized_asset.into_val(&self.env).to_object())
+            .unwrap_infallible()
+            .into_val(&self.env)
+    }
+
+    /// Upload the contract Wasm code to the network.
+    ///
+    /// Returns the hash of the uploaded Wasm that can be then used for
+    /// the contract deployment.
```

Comment:
Can you retain the example we used to have on the `install_contract_wasm` function?

Reply:
Done, though in general doctests here are weird (only the module as a whole is covered and none of the `with_` functions were). We could improve coverage as a followup though.

Reply:
I'd love to get doc tests onto most functions. It's a bit scattered, some types have them, some functions. It may make sense for this example to be moved into the deploy modules doctests examples instead. Happy to figure that out later though, you don't have to address it here, would be good just to keep it for moving later.